manager/assets/modext/widgets/core/modx.combo.js

```diff
@@ -371,28 +371,56 @@ Ext.reg('modx-combo-policy',MODx.combo.Policy);
 
 MODx.combo.Template = function(config) {
     config = config || {};
+
     Ext.applyIf(config,{
-        name: 'template'
-        ,hiddenName: 'template'
-        ,displayField: 'templatename'
-        ,valueField: 'id'
-        ,pageSize: 20
-        ,fields: ['id','templatename','description','category_name']
-        ,tpl: new Ext.XTemplate('<tpl for="."><div class="x-combo-list-item"><span style="font-weight: bold">{templatename:htmlEncode}</span>'
-            ,'<tpl if="category_name"> - <span style="font-style:italic">{category_name:htmlEncode}</span></tpl>'
-            ,'<br />{description:htmlEncode()}</div></tpl>')
-        ,url: MODx.config.connector_url
-        ,baseParams: {
-            action: 'element/template/getlist'
-            ,combo: 1
-        }
-        // ,listWidth: 350
-        ,allowBlank: true
+        url         : MODx.config.connector_url,
+        baseParams  : {
+            action      : 'element/template/getlist',
+            combo       : 1,
+            limit       : 0
+        },
+        fields      : ['id', 'templatename', 'description', 'category_name', 'preview', 'time'],
+        name        : 'template',
+        hiddenName  : 'template',
+        displayField : 'templatename',
+        valueField  : 'id',
+        pageSize    : 20,
+        allowBlank  : true,
+        editable    : true,
+        typeAhead   : true,
+        tpl         : new Ext.XTemplate('<tpl for=".">' +
+            '<tpl if="!Ext.isEmpty(this.getGroup(values.category_name, values.time))">' +
+                '<div class="x-combo-list-group">{this.label}</div>' + 
```

Comment:
As the `label` appears to be coming from a remote source , please add `htmlEncode` as filter to it to prevent persistent XSS.